### DISCUSSION TITLE
AI Code Reviewer

### DISCUSSION THREAD

**Discussion by julialawrence** [ministryofjustice/analytical-platform#2642] [Firebreak Winter 23]
Idea Description:
Implement an AI tool that assists in Terraform and Python code reviews, integrating it with our release process. The AI can suggest improvements, flag potential issues, and predict the impact of code changes. 

Why Should We Do This:
This integration will streamline our code review process, leading to higher code quality and faster development cycles. It will also provide a learning opportunity for the team to understand how AI can aid in software development, improving our skills in AI integration and code quality assurance. It'd be an evolution on our current static code analysis checks.

Definition of Done:
A Github workflow in one of our public repos that reviews PRs and outputs a list of suggestions.

What Skills Do We Need?
Ops and Dev
AWS
Experience or desire to learn GHA

**Comment by bagg3rs** [comment]
Love it, tide us over until we can have some [GitHub Co-Pilot](https://githubnext.com/projects/copilot-for-pull-requests#reviewing-pull-requests-with-ai)
